Add user and group synchronization to Humanitec integration

lordsarcastic opened the pull request:
- Updated README.md to include Users and Groups in the integration overview.
- Enhanced main.py with a new sync_users_and_groups method to fetch and sync users and groups.
- Added get_users_and_groups method in HumanitecClient to retrieve users and groups in a single API call.
- Updated blueprints.json to define schemas for Humanitec User and Group entities.

lordsarcastic committed:
Add user and group synchronization to Humanitec integration

- Updated README.md to include Users and Groups in the integration overview.
- Enhanced main.py with a new sync_users_and_groups method to fetch and sync users and groups.
- Added get_users_and_groups method in HumanitecClient to retrieve users and groups in a single API call.
- Updated blueprints.json to define schemas for Humanitec User and Group entities.

lordsarcastic committed:
Refactor user and group separation logic in HumanitecClient

- Updated the method for separating users and groups to use a single loop for improved efficiency and readability.

mk-armah commented on integration/clients/humanitec_client.py:
```diff
@@ -254,3 +254,23 @@ async def get_deployment_deltas(self, app: Dict[str, Any]) -> List[Dict[str, Any
+            "GET", endpoint, headers=humanitec_headers
+        )
+        
+        # Separate users and groups using a single loop
```
remove comment

mk-armah commented on integration/main.py:
```diff
@@ -556,6 +558,59 @@ def create_deployment_delta_entity(deployment_delta, application):
+    async def sync_users_and_groups(self) -> None:
+        logger.info(f"Syncing entities for blueprints {BLUEPRINT.USER} and {BLUEPRINT.GROUP}")
+        
+        # Single API call to get all users and groups
```
remove comment

mk-armah commented on integration/main.py:
```diff
@@ -556,6 +558,59 @@ def create_deployment_delta_entity(deployment_delta, application):
+            for group in groups
+        ]
+
+        # Execute all tasks together
```
remove comment

mk-armah commented on resources/blueprints.json:
whats the relationship between users and groups ? lets set that in the blueprint and mapping
Ideally, Group -> [User]

lordsarcastic replied:
There is no relationship between them. The API doesn't provide such information

mk-armah reviewed (changes requested):
Left few comments

lordsarcastic committed:
Remove needless comments

lordsarcastic committed:
Enhance user and group synchronization in Humanitec integration

- Refactored sync_users_and_groups method to improve user-group relationship mapping.
- Added get_users_in_group method in HumanitecClient to fetch users by group.
- Updated create_user_entity to include group relations in the user entity.
- Modified blueprints.json to define relations for humanitecGroup.

mk-armah commented on resources/blueprints.json:
```diff
@@ -602,7 +602,14 @@
     "calculationProperties": {},
     "aggregationProperties": {},
-    "relations": {}
+    "relations": {
+      "humanitecGroup": {
+        "title": "Groups",
+        "target": "humanitecGroup",
+        "required": false,
+        "many": true
+      }
+    }
```
relationship is typically the reverse of this

`Group -> [User]`, instead of `User -> [Groups]`

mk-armah reviewed (approved):
LGTM
